R/pool.R

```diff
@@ -45,118 +53,178 @@ Pool <- R6Class("Pool",
       }
       reg.finalizer(self,
         function(self) {
-          protectDefaultScheduler({
-            freeEnv <- private$freeObjects
-            if (length(freeEnv) > 0) {
-              self$close()
-            }
-          })
+          if (self$valid) {
+            warning("Closing leaked pool.")
+            self$close()
+          }
         },
         onexit = TRUE)
```

Comment:
I personally lean towards `onexit = TRUE` because I viscerally don't like the idea of not cleaning up (at least in code, my apartment might disagree with me). But I see why that could be annoying. So, I'm not sure either. If no-one else weighs in, I think we should follow whatever you feel is better, because my thoughts on this are really just personal preference.

(I just realized only the `Pool` object had `onexit = TRUE`, and so for now, I added that to the object's finalizer as well. We can later drop them both if we decide so, but I just wanted to make it consistent, one way or another.)